cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -38,18 +38,18 @@ if(OMR_ENV_DATA64)
 		list(APPEND OMR_PLATFORM_SHARED_LINKER_OPTIONS
 			-m64
 		)
+	elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "x86|i[3-6]86")
```

Comment:
Is this check needed now that you're checking `if(OMR_ARCH_X86)` for the entire block?

If so, `OMR_ENV_DATA32` might be a better variable to check against, that way it reads as

```
if x86:
  if 64-bit build: "-m64"
  else if 32-bit build: "-m32 -msse2"
```

and should also work if you're on 64-bit machine but want to do a 32-bit build.

Reply:
I think you are right, there are no other scenarios on the OMR_ARCH_X86 architecture other than i386 and and64.

Reply:
No magical secret that I know of, I don't know of anyone who's tried cross or is actively working on it at the moment.

Reply:
@ymanton 
> LGTM
Does anything else need to happen on my side before this PR gets merged?

Reply:
I'm OK with the changes. @0xdaryl?